lib/rubocop/ast/node/arg_node.rb

```diff
@@ -0,0 +1,70 @@
+# frozen_string_literal: true
+
+module RuboCop
+  module AST
+    # A node extension for `arg`, `optarg`, `restarg`, `kwarg`, `kwoptarg`,
+    # `kwrestarg`, `blockarg`, `shadowarg` and `forward_arg` nodes.
+    # This will be used in place of a plain node when the builder constructs
+    # the AST, making its methods available to all `arg` nodes within RuboCop.
+    class ArgNode < Node
+      # Returns the name of an argument.
+      #
+      # @return [Symbol, nil] the name of the argument
+      def name
+        node_parts[0]
+      end
+
+      # Returns the default value of the argument, if any.
+      #
+      # @return [Node, nil] the default value of the argument
+      def default_value
+        return unless optional?
+
+        node_parts[1]
+      end
+
+      # Checks whether the argument is optional (it has a default value)
+      #
+      # @return [Boolean] whether the argument is optional
+      def optional?
+        optarg_type? || kwoptarg_type?
+      end
+
+      # Checks whether the argument is a block argument
+      #
+      # @return [Boolean] whether the argument is a block argument
+      def block?
+        blockarg_type?
+      end
+
+      # Checks whether the argument is splatted.
+      #
+      # @return [Boolean] whether the argument is splatted
+      def splatted?
+        restarg_type?
+      end
+
+      # Checks whether the argument is double splatted.
+      #
+      # @return [Boolean] whether the argument is a keyword double splat
+      def double_splatted?
+        kwrestarg_type?
+      end
+      alias keyword? double_splatted?
```

Comment:
I don't understand `keyword?` here... What about `kwarg` and `kwoptarg`?

Reply:
Yeah I think I was mid-thought and got distracted.